Docs(Metrics): 21.03 feature documentation for Transaction and Raft metrics (#78)
dgraph-io/dgraph-docs: content/deploy/metrics.md

+++
date = "2017-03-20T22:25:17+11:00"
title = "Metrics"

> description = "Dgraph database helps administrators by providing metrics on Dgraph instance activity, disk activity, server node health, memory and Raft leadership."

Reviewer: ```suggestion
description = "Dgraph database helps administrators by providing metrics on Dgraph instance activity, disk activity, server node health, memory, and Raft leadership."
```

weight = 5
[menu.main]
    parent = "admin"
+++

Dgraph database provides metrics on Dgraph instance activity, disk activity,

> server node health, memory and Raft leadership. It also provides built-in

Reviewer: ```suggestion
server node health, memory, and Raft leadership. It also provides built-in
```

By looking at all three metrics you can see how much memory a Dgraph process is
holding from the operating system and how much is actively in use.

 Metric                          | Description
 -------                          | -----------
 `dgraph_memory_idle_bytes`       | Estimated amount of memory that is being held idle that could be reclaimed by the OS.
 `dgraph_memory_inuse_bytes`      | Total memory usage in bytes (sum of heap usage and stack usage).
 `dgraph_memory_proc_bytes`       | Total memory usage in bytes of the Dgraph process. On Linux/macOS, this metric is equivalent to resident set size. On Windows, this metric is equivalent to [Go's runtime.ReadMemStats](https://golang.org/pkg/runtime/#ReadMemStats).

## Raft leadership metrics

Raft leadership metrics let you track changes in Raft leadership for Dgraph

> Alpha and Dgraph Zero nodes (or *node instances*) in your cluster. These metrics

Reviewer: ```suggestion
Alpha and Dgraph Zero nodes in your cluster. These metrics
```

Reviewer: I don't think we ever use the term "node instances". It's either been one or the other.

Author: Understood.

include a group label along with the node name, so that you can determine which
metrics apply to which Raft groups. 

Metric                             | Description
-------                            | -----------
`dgraph_raft_has_leader`           | Value is 1 when the node has a leader; otherwise 0.

> `dgraph_raft_is_leader`            | Value is 1 when the node is the leader if its group; otherwise 0.

Reviewer: ```suggestion
`dgraph_raft_is_leader`            | Value is 1 when the node is the leader of its group; otherwise 0.
```

> `dgraph_raft_leader_changes_total` | Incremented each time the leadership of the node's group changes.

Reviewer: ```suggestion
`dgraph_raft_leader_changes_total` | The total number of leader changes seen by this node
```

Author: Fixed.
